R/dash.R

```diff
@@ -1056,6 +1087,41 @@ Dash <- R6::R6Class(
       self$config$silence_routes_logging <- dev_tools_silence_routes_logging
       self$config$props_check <- dev_tools_props_check
 
+      if (private$debug && self$config$ui) {
+        self$server$on('before-request', function(server, ...) {
+          self$server$set_data("timing_information", list(
+            "__dash_server" = list(
+              "dur" = as.numeric(Sys.time()),
+              "desc" = NULL
+            )
+          ))
+        })
+
+        self$server$on('request', function(server, request, ...) {
+          timing_information <- self$server$get_data('timing-information')
+          dash_total <- timing_information[['__dash_server']]
+          dash_total[['dur']] <- round(as.numeric(Sys.time() - dash_total[['dur']]) * 1000)
+          
+          request$response$append_header('Server-Timing', 
```

Comment:
The `request` event isn't passed a `response` object here, because [it can be retrieved/modified via `request`](https://www.data-imaginist.com/2017/when-a-fire-starts-to-burn/):

> _In the same vein, the return values of request event handlers are ignored and the response is not passed to after-request event handlers as the response can be extracted directly from the request._